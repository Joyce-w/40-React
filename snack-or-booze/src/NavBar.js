import React from "react";
import "./NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";

function NavBar() {
  return (
    <div>
      <Navbar expand="md">
        <NavLink exact to="/" className="navbar-brand">
          
          Snack or Booze
        </NavLink>

        <Nav className="ml-auto" navbar>
            <NavLink to="/snacks">Snacks</NavLink>
            <NavLink to="/drinks">Drinks</NavLink>
            <NavLink to="/newItem">New Item</NavLink>
        </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;
