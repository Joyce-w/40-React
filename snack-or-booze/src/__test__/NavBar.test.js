import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import NavBar from '../NavBar'
import {MemoryRouter} from 'react-router-dom'

//smoke test
it("should render w/o crashing", () => {
    render(<MemoryRouter><NavBar /></MemoryRouter>)
})

//snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<MemoryRouter><NavBar /></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot();
})

//specialized tests
test("Should render NavBar text", function () {
    //render uses 2 () to render + write on multiple lines
    const { getByText } = render((
        <MemoryRouter initialEntries={['/']}>
            <NavBar />
        </MemoryRouter>
    ));
    expect(getByText(`Snack or Booze`)).toBeInTheDocument();
});

