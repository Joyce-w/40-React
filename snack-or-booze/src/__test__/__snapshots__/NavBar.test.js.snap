// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <a
          href="/snacks"
        >
          Snacks
        </a>
        <a
          href="/drinks"
        >
          Drinks
        </a>
        <a
          href="/newItem"
        >
          New Item
        </a>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
